examples/core-triangle/App.zig

```diff
@@ -24,64 +25,95 @@ pub fn init(
+        const color_target = gpu.ColorTargetState{
+            .format = core.windows.get(core.main_window, .framebuffer_format),
+            .blend = &blend,
+        };
+
+        // Fragment state describes which shader and entrypoint to use for rendering fragments.
+        const fragment = gpu.FragmentState.init(.{
             .module = shader_module,
-            .entry_point = "vertex_main",
-        },
-    };
-    const pipeline = core.device.createRenderPipeline(&pipeline_descriptor);
-
-    // Store our render pipeline in our module's state, so we can access it later on.
-    app.* = .{
-        .title_timer = try mach.time.Timer.start(),
-        .pipeline = pipeline,
-    };
-
-    // TODO(object): window-title
-    // try updateWindowTitle(core);
+            .entry_point = "frag_main",
+            .targets = &.{color_target},
+        });
+
+        // Create our render pipeline that will ultimately get pixels onto the screen.
+        const label = @tagName(mach_module) ++ ".init";
+        const pipeline_descriptor = gpu.RenderPipeline.Descriptor{
+            .label = label,
+            .fragment = &fragment,
+            .vertex = gpu.VertexState{
+                .module = shader_module,
+                .entry_point = "vertex_main",
+            },
+        };
+        const pipeline = main_window.device.createRenderPipeline(&pipeline_descriptor);
+
+        // Store our render pipeline in our module's state, so we can access it later on.
+        app.* = .{
+            .title_timer = try mach.time.Timer.start(),
+            .pipeline = pipeline,
+        };
+    }
 }
 
+// TODO(object): window-title
+// try updateWindowTitle(core);
+
 pub fn tick(app: *App, core: *mach.Core) void {
     while (core.nextEvent()) |event| {
         switch (event) {
+            .key_press => |ev| {
+                switch (ev.key) {
+                    .right => {
+                        var w = core.windows.getValue(core.main_window);
+                        w.width = w.width + 10;
+                        core.windows.setValue(core.main_window, w);
+                    },
+                    .left => {
+                        var w = core.windows.getValue(core.main_window);
+                        w.width = w.width - 10;
+                        core.windows.setValue(core.main_window, w);
+                    },
+                    .up => {
+                        core.windows.set(core.main_window, .height, core.windows.get(core.main_window, .height) + 10);
+                    },
+                    .down => {
+                        core.windows.set(core.main_window, .height, core.windows.get(core.main_window, .height) - 10);
+                    },
+                    else => {},
+                }
+            },
             .close => core.exit(),
             else => {},
         }
     }
 
+    var main_window = core.windows.getValue(core.main_window);
+
+    // Window is ready when device is not null
```

Comment:
no longer relevant right?

Reply:
I reverted removing main_window because without creating the window in core.init, we pass that along to the user to create in app.init, but the window wont be useable until later on, so they cant really init without the device.

Reply:
did we have a solution for this right now?

Reply:
Sorry, I mean the comment `Window is ready when device is not null` is not relevant anymore, because `device` cannot be null anymore.

Also I think that the if conditional below this line is useless?